### ISSUE TITLE
How to create Visual Studio Code Test package for automated testing of AL extensions?

### ISSUE THREAD

**Issue by sainath654** [Microsoft/AL#4839]
Hello,

I have a created an AL extension APP for Business Central. Now i am in the process of submitting APP to Business Central app source. I have stuck in the place where i need to submit the App Test Automation. 

![123](https://user-images.githubusercontent.com/44765649/55551475-86812700-56f8-11e9-9d2c-dbd7cc6cd0f0.png)
Please guide me how to do that.

Thanks & Regards,
Sainath.

**Comment by ryanweigel** [comment]
Hello,
You need to create a separate .app file that contains all your test codeunits. This test app needs to depend on your regular app (set in manifest/json). That test .app file then goes into that field. We then run these test .app files in our internal build systems for maintenance.

You can find information on creating this test app file here https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/devenv-extension-advanced-example-test

Ryan

**Comment by JohanStenberg100** [comment]
@sainath654 thanks for filing this issue. Ryan's answer provides information how you can unblock yourself. I will now close this issue, please comment here if you still are stuck. If you experience another issue please open another GitHub issue.
